workspace-manager-clienttests/src/main/java/scripts/testscripts/GetWorkspace.java

```diff
@@ -1,78 +1,47 @@
-        WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
-        try {
-            CreateWorkspaceRequestBody requestBody = new CreateWorkspaceRequestBody();
-            requestBody.setId(id);
-            workspace = workspaceApi.createWorkspace(requestBody);
-        } catch (ApiException apiEx) {
-            logger.debug("Caught exception creating workspace ", apiEx);
-        }
-
-        int httpCode = workspaceApi.getApiClient().getStatusCode();
-        logger.info("CREATE workspace HTTP code: {}", httpCode);
-        assertThat(httpCode, equalTo(200));
+    public void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+        throws ApiException {
+        final UUID workspaceId = UUID.randomUUID();
+        final CreateWorkspaceRequestBody requestBody = new CreateWorkspaceRequestBody();
+        requestBody.setId(workspaceId);
+        workspace = workspaceApi.createWorkspace(requestBody);
     }
 
     @Override
-    public void userJourney(TestUserSpecification testUser) throws Exception {
-        ApiClient apiClient = WorkspaceManagerServiceUtils.getClientForTestUser(testUser, server);
-        WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
-
+    public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspaceApi)
+        throws ApiException {
         /**
          * This GetWorkspace test expects a valid workspace id
          * created by the setup step.
          *
          * Throw exception if anything goes wrong
          * **/
-        WorkspaceDescription workspaceDescription = workspaceApi.getWorkspace(workspace.getId());
-        int httpCode = workspaceApi.getApiClient().getStatusCode();
-        logger.info("GET workspace HTTP code: {}", httpCode);
-        assertThat(httpCode, equalTo(200));
+        final WorkspaceDescription workspaceDescription = workspaceApi.getWorkspace(workspace.getId());
     }
 
     @Override
-    public void cleanup(List<TestUserSpecification> testUsers) throws Exception {
-        assertThat("There must be at least one test user in configs/testusers directory.", testUsers!=null && testUsers.size()>0);
-        ApiClient apiClient = WorkspaceManagerServiceUtils.getClientForTestUser(testUsers.get(0), server);
-        WorkspaceApi workspaceApi = new WorkspaceApi(apiClient);
-
-        try {
-            workspaceApi.deleteWorkspace(workspace.getId());
-        } catch (ApiException apiEx) {
-            logger.debug("Caught exception deleting workspace ", apiEx);
-        }
-
-        int httpCode = workspaceApi.getApiClient().getStatusCode();
-        logger.info("DELETE workspace HTTP code: {}", httpCode);
-        assertThat(httpCode, equalTo(204));
+    public void doCleanup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+        throws ApiException {
+        workspaceApi.deleteWorkspace(workspace.getId());
```

Comment:
I like the terser code. In the edge case of an error deleting the workspace, do the logs generated by the terser code give us enough detail to debug the error?

Reply:
They should give the whole exception. Since it's a void method, I don't know what else to check, other than re-checking for the workspace.